sql/2024/sustainability/cache_header_usage.sql

```diff
@@ -0,0 +1,50 @@
+#standardSQL
+# The distribution of cache header adoption on websites by client.
+
+SELECT
+  client,
+  COUNT(0) AS total_requests,
+
+  COUNTIF(uses_cache_control) AS total_using_cache_control,
+  COUNTIF(uses_max_age) AS total_using_max_age,
+  COUNTIF(uses_expires) AS total_using_expires,
+  COUNTIF(uses_max_age AND uses_expires) AS total_using_max_age_and_expires,
+  COUNTIF(uses_cache_control AND uses_expires) AS total_using_both_cc_and_expires,
+  COUNTIF(NOT uses_cache_control AND NOT uses_expires) AS total_using_neither_cc_and_expires,
+  COUNTIF(uses_cache_control AND NOT uses_expires) AS total_using_only_cache_control,
+  COUNTIF(NOT uses_cache_control AND uses_expires) AS total_using_only_expires,
+
+  COUNTIF(uses_cache_control) / COUNT(0) AS pct_cache_control,
+  COUNTIF(uses_max_age) / COUNT(0) AS pct_using_max_age,
+  COUNTIF(uses_expires) / COUNT(0) AS pct_using_expires,
+  COUNTIF(uses_max_age AND uses_expires) / COUNT(0) AS pct_using_max_age_and_expires,
+  COUNTIF(uses_cache_control AND uses_expires) / COUNT(0) AS pct_using_both_cc_and_expires,
+  COUNTIF(NOT uses_cache_control AND NOT uses_expires) / COUNT(0) AS pct_using_neither_cc_nor_expires,
+  COUNTIF(uses_cache_control AND NOT uses_expires) / COUNT(0) AS pct_using_only_cache_control,
+  COUNTIF(NOT uses_cache_control AND uses_expires) / COUNT(0) AS pct_using_only_expires
+
+FROM (
+  SELECT
+    client,
+
+    JSON_EXTRACT_SCALAR(summary, '$.resp_expires') IS NOT NULL AND TRIM(JSON_EXTRACT_SCALAR(summary, '$.resp_expires')) != '' AS uses_expires,
+    JSON_EXTRACT_SCALAR(summary, '$.resp_cache_control') IS NOT NULL AND TRIM(JSON_EXTRACT_SCALAR(summary, '$.resp_cache_control')) != '' AS uses_cache_control,
+    REGEXP_CONTAINS(JSON_EXTRACT_SCALAR(summary, '$.resp_cache_control'), r'(?i)max-age\s*=\s*[0-9]+') AS uses_max_age,
+
+    JSON_EXTRACT_SCALAR(summary, '$.resp_etag') IS NULL OR TRIM(JSON_EXTRACT_SCALAR(summary, '$.resp_etag')) = '' AS uses_no_etag,
+    JSON_EXTRACT_SCALAR(summary, '$.resp_etag') IS NOT NULL AND TRIM(JSON_EXTRACT_SCALAR(summary, '$.resp_etag')) != '' AS uses_etag,
+    JSON_EXTRACT_SCALAR(summary, '$.resp_last_modified') IS NOT NULL AND TRIM(JSON_EXTRACT_SCALAR(summary, '$.resp_last_modified')) != '' AS uses_last_modified,
+
+    REGEXP_CONTAINS(TRIM(JSON_EXTRACT_SCALAR(summary, '$.resp_etag')), '^W/".*"') AS uses_weak_etag,
+    REGEXP_CONTAINS(TRIM(JSON_EXTRACT_SCALAR(summary, '$.resp_etag')), '^".*"') AS uses_strong_etag
+
+  FROM
+    `httparchive.all.requests`
+  WHERE
+   date = '2024-06-01'
```

Comment:
I'm wondering if we should filter by `is_root_page = TRUE`.
I'm still not sure when we should analyze data per page level or site level. I see that some of our queries filter that to analyze site level but other like this don't.
If we don't wanna filter it we might need to rephrase a few things in the Google doc to say "pages" instead of "websites" like what was suggested in this [slack message](https://httparchive.slack.com/archives/CGFLV2S06/p1725518749744479)

Reply:
As this particular query is requests I think this is OK. Resolving.